dining/views.py

```diff
@@ -0,0 +1,90 @@
+from datetime import datetime
+
+from django.http import HttpResponseForbidden, JsonResponse
+from legacy.models import Account, DiningBalance, DiningTransaction
+from options.models import get_option, get_value
+from rest_framework.views import APIView
+
+from studentlife.utils import date_iso_eastern, get_new_start_end
+
+
+# Create your views here.
+
+
+class Dashboard(APIView):
+    def get(self, request, format=None):
+        if not self.request.user.is_authenticated:
+            return HttpResponseForbidden()
+
+        self.pennid = "76627463"  # request.user.username
+        self.uid = Account.objects.filter(pennid=self.pennid)[0].id
+        json = self.balance()
+
+        start, end = self.get_semester_start_end()
+        json["start-of-semester"] = date_iso_eastern(start)
+        json["end-of-semester"] = date_iso_eastern(end)
+
+        json["cards"] = {"recent-transactions": self.recent_transactions_card()}
+
+        return JsonResponse(json)
+
+    def balance(self):
+        balance = DiningBalance.objects.filter(account=self.uid).order_by("-created_at")
```

Comment:
Like I mentioned above, I think it would be better to pass `uid` into this function instead of setting it on `self` and then referencing the instance.